qa/rpc-tests/wallet_orchard_change.py

```diff
@@ -0,0 +1,123 @@
+
+    def run_test(self):
+        # Sanity-check the test harness.
+        assert_equal(self.nodes[0].getblockcount(), 200)
+
+        # Create an account with funds in the Sapling receiver.
+        acct0 = self.nodes[0].z_getnewaccount()['account']
+        ua0 = self.nodes[0].z_getaddressforaccount(acct0, ['sapling', 'orchard'])['address']
+
+        recipients = [{"address": ua0, "amount": 10}]
+        myopid = self.nodes[0].z_sendmany(get_coinbase_address(self.nodes[0]), recipients, 1, 0, 'AllowRevealedSenders')
+        wait_and_assert_operationid_status(self.nodes[0], myopid)
+
+        # Mine the tx & activate NU5.
+        self.sync_all()
+        self.nodes[0].generate(5)
+        self.sync_all()
+
+        assert_equal(
+            {'pools': {'sapling': {'valueZat': 10_0000_0000}}, 'minimum_confirmations': 1},
+            self.nodes[0].z_getbalanceforaccount(acct0),
+        )
+
+        # We want to generate an Orchard change note on node 0. We do this by
+        # sending funds to an Orchard-only UA on node 1.
+        acct1 = self.nodes[1].z_getnewaccount()['account']
+        ua1 = self.nodes[1].z_getaddressforaccount(acct1, ['orchard'])['address']
+
+        recipients = [{"address": ua1, "amount": 1}]
+        myopid = self.nodes[0].z_sendmany(ua0, recipients, 1, 0)
+        source_tx = wait_and_assert_operationid_status(self.nodes[0], myopid)
+
+        self.sync_all()
+        self.nodes[0].generate(1)
+        self.sync_all()
+
+        # The nodes have the expected split of funds.
+        assert_equal(
+            {'pools': {'orchard': {'valueZat': 9_0000_0000}}, 'minimum_confirmations': 1},
+            self.nodes[0].z_getbalanceforaccount(acct0),
+        )
+        assert_equal(
+            {'pools': {'orchard': {'valueZat': 1_0000_0000}}, 'minimum_confirmations': 1},
+            self.nodes[1].z_getbalanceforaccount(acct1),
+        )
+
+        # The Orchard note for node 0 is a change note.
+        self.check_has_output(0, source_tx, {
+            'pool': 'orchard',
+            'outgoing': False,
+            'walletInternal': True,
+            'value': 9,
+            'valueZat': 9_0000_0000,
+        })
+
+        # Shut down the nodes, and restart so that we can check wallet load
+        stop_nodes(self.nodes)
+        wait_bitcoinds()
+        self.setup_network()
+
+        # The node have unaltered balances.
+        assert_equal(
+            {'pools': {'orchard': {'valueZat': 9_0000_0000}}, 'minimum_confirmations': 1},
+            self.nodes[0].z_getbalanceforaccount(acct0),
+        )
+        assert_equal(
+            {'pools': {'orchard': {'valueZat': 1_0000_0000}}, 'minimum_confirmations': 1},
+            self.nodes[1].z_getbalanceforaccount(acct1),
+        )
+
+        # Send another Orchard transaction from node 0 to node 1.
+        myopid = self.nodes[0].z_sendmany(ua0, recipients, 1, 0)
```

Comment:
This is actually sending back to `ua0`, but the comment says it is sending to node 1. I don't think it matters which, but the comment and code should be consistent.